extension/duckdb/src/duckdb_storage.cpp

```diff
@@ -17,26 +18,19 @@ static std::string getCatalogNameFromPath(const std::string& dbPath) {
     return path.stem().string();
 }
 
-static void validateDuckDBPathExistence(const std::string& dbPath, main::ClientContext* context) {
-    auto vfs = std::make_unique<common::VirtualFileSystem>();
-    if (!vfs->fileOrPathExists(dbPath, context)) {
-        throw common::RuntimeException{
-            common::stringFormat("'{}' is not a valid duckdb database path.", dbPath)};
-    }
-}
-
 std::unique_ptr<main::AttachedDatabase> attachDuckDB(std::string dbName, std::string dbPath,
     main::ClientContext* clientContext, const binder::AttachOption& attachOption) {
     auto catalogName = getCatalogNameFromPath(dbPath);
     if (dbName == "") {
         dbName = catalogName;
     }
-    validateDuckDBPathExistence(dbPath, clientContext);
-    auto duckdbCatalog =
-        std::make_unique<DuckDBCatalog>(dbPath, catalogName, clientContext, attachOption);
+    auto connector = DuckDBConnectorFactory::getDuckDBConnector(dbPath);
+    connector->connect(dbPath, catalogName, clientContext);
+    auto duckdbCatalog = std::make_unique<DuckDBCatalog>(std::move(dbPath), std::move(catalogName),
+        DuckDBStorageExtension::DEFAULT_SCHEMA_NAME, clientContext, *connector, attachOption);
     duckdbCatalog->init();
```

Comment:
You can consider move the call of `init()` inside `DuckDBCatalog` constructor, since you seem to always init after construction? Same for postgres and sqlite.

Reply:
I think it is not a good practice to call a function which does lots of work inside the constructor according to the cpp standard.